models/practices-engineering/common/data_preparation.py

```diff
@@ -0,0 +1,17 @@
+from pathlib import Path
+from typing import List
+
+import pandas as pd
+from sklearn.model_selection import train_test_split
+
+
+def prepare_training_data(
+        csv_path: Path, test_size: float = 0.2, random_state: int = 0
+) -> List:
+    path = Path(csv_path) if isinstance(csv_path, str) else csv_path
```

Comment:
Optional improvement

```suggestion
    path = Path(csv_path)
```

You can pass a `Path` instance to the `Path`  constructor.